src/launcher.ts

```diff
@@ -326,6 +327,24 @@ class Launcher {
         return NOOP_INSTANCE;
     }
 
+    private async runDesktopPerformance(build: IBuild): Promise<IInstance> {
+        const executable = await builds.getBuildExecutable(build);
+
+        const executableExists = await exists(executable);
+        if (!executableExists) {
+            throw new Error(`[build] unable to find executable ${executable} on disk. Is the archive corrupt?`);
+        }
```

Comment:
Maybe we could extract this into a method because the same code seems duplicate later in the file?

Reply:
✅